### PR TITLE
✨ Environment Variable Support

### PR DESCRIPTION
**Pull request by segersniels**
### Added
- Allows user to provide `user.cfg` settings by configuring environment variables

### Changed
- Will no longer exit script if no `user.cfg` is found and will try to fully rely on env vars and default config
- Moved all default fallbacks to DEFAULT section of config 

```
SUPPORTED_COIN_LIST: "XLM TRX ICX EOS IOTA ONT QTUM ETC ADA XMR DASH NEO ATOM DOGE VET BAT OMG BTT"
BRIDGE_SYMBOL: USDT
CURRENT_COIN_SYMBOL:
API_KEY:
API_SECRET_KEY:
```

### CODE REVIEW

**Comment by segersniels** [comment]
Still troubleshooting some minor issues.

**Comment by edeng23** [comment]
Can you please rebase? The config code was moved to a separate file in #148. Thanks!

**Comment by segersniels** [comment]
> Can you please rebase? The config code was moved to a separate file in #148. Thanks!

🚀

File: README.md [hunk 1]
```diff
@@ -54,6 +54,21 @@ Create a .cfg file named `user.cfg` based off `.user.cfg.example`, then add your
 - **scout_transaction_fee** - The transaction fee percentage. This value should be changed, for example, if you are [using BNB to pay for fees](https://www.binance.com/en/support/faq/115000583311-Using-BNB-to-Pay-for-Fees). 
 - **scout_multiplier** - Controls the value by which the difference between the current state of coin ratios and previous state of ratios is multiplied. For bigger values, the bot will wait for bigger margins to arrive before making a trade. 
 
+#### Environment Variables
+
+Most of the options provided in `user.cfg` can also be configured using environment variables.
```
Comment (edeng23) [inline README.md]: Why **most** and not **all**?  Why isn't tld in your list?

File: binance_trade_bot/config.py [hunk 2]
```diff
@@ -4,47 +4,72 @@
...
+        self.BINANCE_API_KEY = os.environ.get("API_KEY") or config.get(
+            USER_CFG_SECTION, "api_key"
+        )
+        self.BINANCE_API_SECRET_KEY = os.environ.get("API_SECRET_KEY") or config.get(
+            USER_CFG_SECTION, "api_secret_key"
+        )
+        self.BINANCE_TLD = os.environ.get("TLD") or config.get(USER_CFG_SECTION, "tld")
 
-        self.SUPPORTED_COIN_LIST = []
+        self.SUPPORTED_COIN_LIST = os.environ.get("SUPPORTED_COIN_LIST").split() or []
```
Comment (edeng23) [inline binance_trade_bot/config.py]: This won't work for people not using environment variables. get() will return Nonetype object, which you cannot split, resulting in an Attribute Error.

**Review by edeng23 — CHANGES_REQUESTED** [review]
You must take into consideration the scenario in which a user (me, for example) does not use environment variables. I'd really appreciate if you can take the time to run the code on a config file. Thanks :)

File: README.md (hunk 1 above)
Comment (segersniels) [reply README.md]: Will add all supported ENV vars to the list. It's a bit outdated

**Comment by segersniels** [comment]
> You must take into consideration the scenario in which a user (me, for example) does not use environment variables. I'd really appreciate if you can take the time to run the code on a config file. Thanks :)

Converted my own running bot to run against config files instead of env vars using the following minimal setup (obviously left out some stuff). Can confirm it still works with just using config files. @edeng23 I would advise that you give it a go using my image to see if you find any issues. Also no issues switching from config to env vars, so people should be able to adapt easily.

### `docker-compose.yml`
```yaml
version: "3"
  
services:
  binance:
    image: segersniels/binance-trade-bot:latest
      - /foo/binance/user.cfg:/app/user.cfg
      - /foo/binance/supported_coin_list:/app/supported_coin_list
    restart: always
```

### Output

```                                                                                                                                                                                                                                                  
$ docker logs   
2021-03-04 11:02:32,267 - crypto_trader_logger - INFO - Starting
2021-03-04 11:02:32,868 - crypto_trader_logger - INFO - Creating database schema if it doesn't already exist
                                                                                                                                                                                                                                                        
$ cat /app/user.cfg                  
[binance_user_config]
api_key=
api_secret_key=
current_coin=
bridge=USDT
tld=com
hourToKeepScoutHistory=1
scout_transaction_fee=0.001
scout_multiplier=5
scout_sleep_time=5
                                                                                                                                                                                                                                                    
$ cat /app/supported_coin_list
BTC
ETH
ADA
BNB
LTC
XMR
XLM
ATOM
EOS
TRX
IOTA
VET
NEO
DASH 
ETC 
NANO 
ICX 
SUSHI 
ALGO 
UNI 
CAKE
```

**Review by edeng23 — APPROVED** [review]

**Pull request merged**